pkg/syncer/spec/mutators/interface.go

```diff
@@ -0,0 +1,27 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mutators
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type Mutator interface {
+	GVR() schema.GroupVersionResource
```

Comment:
am not a fan of this interface. It was kind of legacy already in the syncer. Remove the GVR() here and add that to the registration of the mutator in some GVR->mutator map.

Reply:
Also go doc.